### DISCUSSION TITLE
Is there useIsLoading solution?

### DISCUSSION THREAD

**Discussion by nezriffic** [TanStack/query#3564] [Q&A] [answered]
Hey Team!
I am looking for a global isLoading solution.
`useIsFetching` is great, but I need a version that uses `isLoading` instead of `isFetching` (eg. to show global spinner when something is loading but NOT re-fetching).

Is there such solution maybe?
Thanks.

**Comment by TkDodo** [comment, marked as answer]
1) `suspense` is made for that
2) `useIsFetching({ predicate: (query) => query.state === 'loading' })` should achieve that

**Reply by nezriffic** [reply]
`useIsFetching({
  predicate: (query) => query.state.status === 'loading'
})`

does the trick! Thanks a lot!
